packages/filebrowser/src/listing.ts

```diff
@@ -1123,24 +1123,59 @@ export class DirListing extends Widget {
     if (!files || files.length === 0) {
       return;
     }
-    const length = event.dataTransfer?.items.length;
-    if (!length) {
-      return;
-    }
-    for (let i = 0; i < length; i++) {
-      let entry = event.dataTransfer?.items[i].webkitGetAsEntry();
-      if (entry.isDirectory) {
-        console.log('currently not supporting drag + drop for folders');
-        void showDialog({
-          title: this._trans.__('Error Uploading Folder'),
-          body: this._trans.__(
-            'Drag and Drop is currently not supported for folders'
-          ),
-          buttons: [Dialog.cancelButton({ label: this._trans.__('Close') })]
-        });
+    event.preventDefault();
+
+    if (event.dataTransfer && event.dataTransfer.items) {
+      let items = event.dataTransfer.items;
+
+      const addDirectory = (item: any) => {
+        if (item.isDirectory) {
+          this._model.manager
+            .newUntitled({
+              path: this._model.path,
+              type: 'directory'
+            })
+            .then(async model => {
+              await this._manager.rename(
+                `${this._model.path}/${model.name}`,
+                `${this._model.path}/${item}`
+              );
+              this._model
+                .cd(`${this._model.path}/${item}`)
```

Comment:
You are making a great progress! When I tried it on binder it renamed the directory to `"[object DirectoryEntry]` which highlights that the second argument includes an object rather than a string. This should fix it:

```suggestion
                `${this._model.path}/${item.name}`
              );
              this._model
                .cd(`${this._model.path}/${item.name}`)
```

Reply:
I tried doing the suggested changes. But it is still not working as expected. 
It is changing the current directory, but files and nested files are getting uploaded in the root directory only. 
I am trying to figure out the catch.